chalice/deploy/deployer.py

```diff
@@ -324,18 +325,19 @@ def _first_time_lambda_create(self, config, function_name, stage_name):
             config.project_dir)
         zip_contents = self._osutils.get_file_contents(
             zip_filename, binary=True)
-        return self._aws_client.create_function(
-            function_name, role_arn, zip_contents,
-            config.environment_variables,
-            config.lambda_python_version,
-            self._function_tags(config, stage_name),
-        )
-
-    def _function_tags(self, config, stage_name):
-        # type: (Config, str) -> Dict[str, str]
-        tag = 'version=%s:stage=%s:app=%s' % (chalice_version,
-                                              stage_name, config.app_name)
-        return {'aws-chalice': tag}
+        client_kwargs = {
```

Comment:
I'd prefer not to use this style, having the chain of `if <param> is not None...` makes the code harder to read and I don't know how well the type checker check this style.  Why can't we just pass through `config.lambda_timeout` and `config.lambda_memory_size`?  They'll still be `None` if they're not set right?